docs/code_contribution_guidelines.md

```diff
@@ -290,9 +296,29 @@ either a '+' or a ','. This prefix seems minor but can be extremely helpful in
 determining the scope of a commit at a glance, or when bug hunting to find a
 commit which introduced a bug or regression. 
 
+<a name="IdealGitCommitStructure" />
+
+#### 4.5. Ideal Git Commit Structure
+
+Within the project we prefer small, contained commits for a pull request over a
+single giant commit that touches several flies/packages. Ideal commits build on
+their own, in order to facilitate easy usage of tools like `git bisect` to `git
+cherry-pick`. It's preferred that commits contains an isolated change in a
+single package. In this case, the commit header message should begin with the
+prefix of the modified package. For example, if a commit was made to modify the
+`lnwallet` package, it should start with `lnwallet: `. 
+
+In the case of changes that only build in tandem with changes made in other
+packages, it is permitted for a single commit to be made which contains several
+prefixes such as: `lnwallet+htlcswitch`. This prefix structure along with the
+requirement for atomic contained commits (when possible) make things like
+scanning the set of commits and debugging easier. In the case of a changes that
+touches several packages, and can only compile with the change across several
+packages, a `multi: ` prefix should be used.
+
 <a name="CodeSpacing" />
 
-#### 4.5. Code Spacing 
+#### 4.6. Code Spacing 
```

Comment:
Add guidelines on the following?
* New line after wrapped line. (That didn't end with a closing parenthesis?)
* New line before `select` / `switch` cases
* ... ?

Reply:
Added the second, not sure what you mean by the first, but feel free to make a follow up PR with that if this goes in before then.

Reply:
```
func bla(s int) (
     error) {
<newline here>
    var a int
}

Reply:
Thanks, added!